File: 2873_maximum_value_of_an_ordered_triplet_i.py
class Solution(object):
    def maximumTripletValue(self, v):
        ans, t = 0, 0
        s = [0] * len(v)
        for i in range(len(v) - 2, -1, -1): s[i] = max(s[i + 1], v[i + 1])
        for j in range(1, len(v) - 1):
            t = max(t, v[j - 1])
            ans = max(ans, (t - v[j]) * s[j])
        return ans

File: 2873_maximum_value_of_an_ordered_triplet_i.cpp
class Solution {
public:
    long long maximumTripletValue(vector<int>& z) {
        long long mx = 0;
        for (int x = 0; x < z.size(); x++)
            for (int y = x + 1; y < z.size(); y++)
                for (int w = y + 1; w < z.size(); w++)
                    mx = max(mx, (long long)(z[x] - z[y]) * z[w]);
        return mx;
    }
};
